package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -170,7 +170,26 @@ append_address() {
 }
 
 append_ipset() {
-	xappend "--ipset=$1"
+	if [ -x /sbin/fw4 ]; then
+		local domainlist=${1%/*}
+		local setlist=${1##*/}
+		local nftsets=""
+
+		IFS=,
+		for set in $setlist; do
+			local family=$(nft -t list set inet fw4 "$set" 2>&1 | sed -nre 's#^\t\ttype .*\bipv([46])_addr\b.*$#\1#p')
```

Comment:
No need to take care of the sets ip family type. dnsmasq will try to add the IP addresses to all sets, regardless of the IP version defined for the set. Adding an IPv6 to an IPv4 set and vice versa will silently fail. In the end, the sets are populated correctly.

According to the dnsmasq man page, it's the same behaviour as with ipsets.

Once more, you can find such an implementation in my staging tree.